src/functionHandlers/getProfile/getProfile.ts

```diff
@@ -18,6 +19,17 @@ const getIdentifier = async (event: APIGatewayEvent) => {
   if (!id) {
     throw new createHttpError.BadRequest("Identifier is not provided");
   }
+
+  const apiKey = event.headers["x-api-key"] ?? undefined;
+
+  if (apiKey === undefined) {
+    throw new createHttpError.BadRequest("API key header is not provided");
+  }
+
+  if (isEmpty(apiKey)) {
+    throw new createHttpError.BadRequest("API key value should not be empty");
+  }
```

Comment:
As long as it is not in some predefined list, can just throw already right? Also, it doesn't seem like there is any forms of check in here, both are just checking for the presence of the key. 

Since this is a demo, just have a single key `DEMO` or something and check against that.